src/data-pipeline/utils/utils-lambda.ts

```diff
@@ -138,6 +138,9 @@ export class LambdaUtil {
     this.props.sourceS3Bucket.grantRead(lambdaRole, `${this.props.sourceS3Prefix}*`);
     this.props.pipelineS3Bucket.grantReadWrite(lambdaRole, `${this.props.pipelineS3Prefix}*`);
 
+    const ruleConfigDir = getRuleConfigDir(this.props.pipelineS3Prefix, this.props.projectId);
+    this.props.pipelineS3Bucket.grantReadWrite(lambdaRole, `${ruleConfigDir}*`);
```

Comment:
Does it really need write permission?

Reply:
the emr application and job submit lambda share the same rule. the job submit lambda need write role put initial configure rule files to s3 new apps if config files not exist in s3.